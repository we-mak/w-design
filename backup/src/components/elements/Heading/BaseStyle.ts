import { fontWeight } from "styled-system";
import { css } from "../../../utils/styled";

const BaseStyle = css`
  display: block;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-weight: 500;
  ${fontWeight};
`;

export default BaseStyle;
