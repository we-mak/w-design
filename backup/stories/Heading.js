import React from "react";
import { storiesOf } from "@storybook/react";
import { Heading } from "../src/components";

storiesOf("Heading", module)
  .addWithJSX("Default Heading", () => (
    <React.Fragment>
      <Heading.H1> H1 Heading</Heading.H1>
      <Heading.H2> H2 Heading</Heading.H2>
      <Heading.H3> H3 Heading</Heading.H3>
      <Heading.H4> H4 Heading</Heading.H4>
      <Heading.H5> H5 Heading</Heading.H5>
      <Heading.H6>H6 Heading</Heading.H6>
    </React.Fragment>
  ))
  .addWithJSX("Custom component", () => (
    <Heading.H1 customComponent="span">H1 Heading as span tag</Heading.H1>
  ))
  .addWithJSX("Custom Fontweight", () => (
    <Heading.H1 fontWeight={600}> H1 font-weight 600 </Heading.H1>
  ))
  .addWithJSX("Custom color", () => (
    <Heading.H1 color="green"> H1 Heading</Heading.H1>
  ))
  .addWithJSX("Responsive heading", () => (
    <Heading.H1 fontSize={[2, 3, 4, 5]}> H1 Heading</Heading.H1>
  ));
