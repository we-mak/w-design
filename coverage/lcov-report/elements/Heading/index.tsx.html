<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for elements/Heading/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">elements/Heading</a> index.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.31% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>36/39</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>14/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.82% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>9/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.11% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>35/38</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as React from "react";
import styled, { css } from "styled-components";
import { fontWeight, fontSize, color } from "styled-system";
import { CustomTag } from "../../../common/utils/customTag";
import { GlobProps } from "../../../typings";
&nbsp;
const BaseStyle = css`
  display: block;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-weight: ${(props: any) =&gt; props.theme.fontWeights["bold"]};
  color: ${props =&gt; props.theme.colors["N50"]};
  /* Custom style*/
  ${color}
  ${fontWeight}
`;
&nbsp;
const H1 = styled(CustomTag)`
  font-size: ${props =&gt; props.theme.fontSizes[8]};
  ${BaseStyle}
  ${fontSize}
`;
H1.displayName = "H1";
&nbsp;
const H2 = styled(CustomTag)`
  font-size: ${props =&gt; props.theme.fontSizes[6]};
  ${BaseStyle}
  ${fontSize}
`;
H2.displayName = "H2";
&nbsp;
const H3 = styled(CustomTag)`
  font-size: ${props =&gt; props.theme.fontSizes[5]};
  ${BaseStyle}
  ${fontSize}
`;
H3.displayName = "H3";
&nbsp;
const H4 = styled(CustomTag)`
  font-size: ${props =&gt; props.theme.fontSizes[4]};
  ${BaseStyle}
  ${fontSize}
`;
H4.displayName = "H4";
&nbsp;
const H5 = styled(CustomTag)`
  font-size: ${props =&gt; props.theme.fontSizes[3]};
  ${BaseStyle}
  ${fontSize}
`;
H5.displayName = "H5";
&nbsp;
const H6 = styled(CustomTag)`
  font-size: ${props =&gt; props.theme.fontSizes[2]};
  ${BaseStyle}
  ${fontSize}
`;
H6.displayName = "H6";
&nbsp;
interface HeadingProps extends GlobProps {
  children?: React.ReactChild;
}
&nbsp;
class Heading extends React.PureComponent&lt;HeadingProps&gt; {
  static H1 = H1;
  static H2 = H2;
  static H3 = H3;
  static H4 = H4;
  static H5 = H5;
  static H6 = H6;
&nbsp;
<span class="fstat-no" title="function not covered" >  render() {</span>
    const children = <span class="cstat-no" title="statement not covered" >React.Children.map(this.props.children, <span class="fstat-no" title="function not covered" >child =&gt;</span></span>
<span class="cstat-no" title="statement not covered" >      React.cloneElement(child as React.ReactElement&lt;HTMLElement&gt;)</span>
    );
&nbsp;
<span class="cstat-no" title="statement not covered" >    return &lt;React.Fragment {...this.props}&gt;{children}&lt;/React.Fragment&gt;;</span>
  }
}
&nbsp;
export default Heading;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Mar 11 2019 15:51:13 GMT+0700 (Indochina Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
