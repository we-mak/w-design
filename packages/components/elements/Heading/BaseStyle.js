"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
Object.defineProperty(exports, "__esModule", { value: true });
var styled_system_1 = require("styled-system");
var styled_1 = require("../../../utils/styled");
var BaseStyle = styled_1.css(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  display: block;\n  line-height: 1.25;\n  margin-bottom: 0.5em;\n  margin-top: 0;\n  font-weight: 500;\n  ", ";\n"], ["\n  display: block;\n  line-height: 1.25;\n  margin-bottom: 0.5em;\n  margin-top: 0;\n  font-weight: 500;\n  ", ";\n"])), styled_system_1.fontWeight);
exports.default = BaseStyle;
var templateObject_1;
//# sourceMappingURL=BaseStyle.js.map