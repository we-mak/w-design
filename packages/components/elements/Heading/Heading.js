"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) if (e.indexOf(p[i]) < 0)
            t[p[i]] = s[p[i]];
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var clean_tag_1 = require("clean-tag");
var styled_system_1 = require("styled-system");
var theme_1 = require("../../../common/theme");
var styled_1 = require("../../../utils/styled");
var BaseStyle_1 = require("./BaseStyle");
var TagElement = function (_a) {
    var customComponent = _a.customComponent, props = __rest(_a, ["customComponent"]);
    return React.createElement(clean_tag_1.default[customComponent || "div"], __assign({}, props));
};
var colors = theme_1.default.colors, fontSizes = theme_1.default.fontSizes;
// H1
var H1 = styled_1.default(TagElement)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"], ["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"])), fontSizes[8], colors.N90, BaseStyle_1.default, styled_system_1.color, styled_system_1.fontSize);
H1.displayName = "Heading.H1";
// H2
var H2 = styled_1.default(TagElement)(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"], ["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"])), fontSizes[6], colors.N90, BaseStyle_1.default, styled_system_1.color, styled_system_1.fontSize);
H2.displayName = "Heading.H2";
// H3
var H3 = styled_1.default(TagElement)(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"], ["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"])), fontSizes[5], colors.N90, BaseStyle_1.default, styled_system_1.color, styled_system_1.fontSize);
H3.displayName = "Heading.H3";
// H4
var H4 = styled_1.default(TagElement)(templateObject_4 || (templateObject_4 = __makeTemplateObject(["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"], ["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"])), fontSizes[4], colors.N90, BaseStyle_1.default, styled_system_1.color, styled_system_1.fontSize);
H4.displayName = "Heading.H4";
// H5
var H5 = styled_1.default(TagElement)(templateObject_5 || (templateObject_5 = __makeTemplateObject(["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"], ["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"])), fontSizes[3], colors.N70, BaseStyle_1.default, styled_system_1.color, styled_system_1.fontSize);
H5.displayName = "Heading.H5";
// H6
var H6 = styled_1.default(TagElement)(templateObject_6 || (templateObject_6 = __makeTemplateObject(["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"], ["\n  font-size: ", ";\n  color: ", ";\n  ", ";\n  ", ";\n  ", ";\n"])), fontSizes[1], colors.N70, BaseStyle_1.default, styled_system_1.color, styled_system_1.fontSize);
H6.displayName = "Heading.H6";
var Heading = /** @class */ (function (_super) {
    __extends(Heading, _super);
    function Heading() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Heading.prototype.render = function () {
        var children = React.Children.map(this.props.children, function (child) { return React.cloneElement(child); });
        return React.createElement(React.Fragment, __assign({}, this.props), children);
    };
    Heading.H1 = H1;
    Heading.H2 = H2;
    Heading.H3 = H3;
    Heading.H4 = H4;
    Heading.H5 = H5;
    Heading.H6 = H6;
    Heading.displayName = "Heading";
    return Heading;
}(React.Component));
exports.default = Heading;
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5, templateObject_6;
//# sourceMappingURL=Heading.js.map