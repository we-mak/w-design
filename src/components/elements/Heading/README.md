| Props           | type              | default     | description                   |
| --------------- | ----------------- | ----------- | ----------------------------- |
| customComponent | string            | div         | change html tag definition    |
| fontWeight      | string[]/number[] | strong      | change font-weight style      |
| color           | string            | theme color | change color style from theme |

Example `customComponent="span"` to change tag to `span`

```js
<Provider>
  <Heading.H1> H1 Heading</Heading.H1>
  <Heading.H1 customComponent="span">H1 Heading as span tag</Heading.H1>

  <Heading.H2> H2 Heading</Heading.H2>
  <Heading.H3> H3 Heading</Heading.H3>
  <Heading.H4> H4 Heading</Heading.H4>
  <Heading.H5> H5 Heading</Heading.H5>
  <Heading.H6>H6 Heading</Heading.H6>
</Provider>
```

Default font-weight is `500`, it's could be changed by using `fontWeight` prop

```js
<Provider>
  <Heading.H1 fontWeight={600}> H1 font-weight 600 </Heading.H1>
</Provider>
```

## Custom color

Color could be change easily using default or custom theme colors

**Example base on default theme color**

```js static
const theme = {
  colors: {
    // Primary
    B70: "#004660",
    B60: "#007099",
    B50: "#008CC0", // Primary Color
    B40: "#009AD3",
    B30: "#00A7E4",
    B20: "#00B7FA",
    B10: "#3ECBFF",
    B7: "#7CDCFF",
    B5: "#B1EAFF"
  }
};
```

```js
<Provider>
  <Heading.H1 color="B50"> H1 Heading</Heading.H1>
</Provider>
```

## Responsive heading

```js
<Provider>
  <Heading.H1 fontSize={[2, 3, 4, 5]}> H1 Heading</Heading.H1>
</Provider>
```
