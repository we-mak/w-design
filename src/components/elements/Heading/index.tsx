import * as React from "react";
import styled, { css } from "styled-components";
import { fontWeight, fontSize, color } from "styled-system";
import { CustomTag } from "../../../common/utils/customTag";
import { GlobProps } from "../../../typings";
import {
  colors,
  fontWeights,
  fontSizes
} from "../../../common/styleUtils/theme";

const BaseStyle = css`
  display: block;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-weight: ${fontWeights["bold"]};
  color: ${colors["N50"]};
  /* Custom style*/
  ${color}
  ${fontWeight}
`;

const H1 = styled(CustomTag)`
  font-size: ${fontSizes[8]};
  ${BaseStyle}
  ${fontSize}
`;
H1.displayName = "H1";

const H2 = styled(CustomTag)`
  font-size: ${fontSizes[6]};
  ${BaseStyle}
  ${fontSize}
`;
H2.displayName = "H2";

const H3 = styled(CustomTag)`
  font-size: ${fontSizes[5]};
  ${BaseStyle}
  ${fontSize}
`;
H3.displayName = "H3";

const H4 = styled(CustomTag)`
  font-size: ${fontSizes[4]};
  ${BaseStyle}
  ${fontSize}
`;
H4.displayName = "H4";

const H5 = styled(CustomTag)`
  font-size: ${fontSizes[3]};
  ${BaseStyle}
  ${fontSize}
`;
H5.displayName = "H5";

const H6 = styled(CustomTag)`
  font-size: ${fontSizes[2]};
  ${BaseStyle}
  ${fontSize}
`;
H6.displayName = "H6";

interface HeadingProps extends GlobProps {
  children?: React.ReactChild;
}

class Heading extends React.PureComponent<HeadingProps> {
  static H1 = H1;
  static H2 = H2;
  static H3 = H3;
  static H4 = H4;
  static H5 = H5;
  static H6 = H6;

  render() {
    const children = React.Children.map(this.props.children, child =>
      React.cloneElement(child as React.ReactElement<HTMLElement>)
    );

    return <React.Fragment {...this.props}>{children}</React.Fragment>;
  }
}

export default Heading;
