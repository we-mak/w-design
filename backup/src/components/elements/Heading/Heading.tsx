import * as React from "react";
import tag from "clean-tag";
import { color, fontSize } from "styled-system";
import theme from "../../../common/theme";
import styled from "../../../utils/styled";
import BaseStyle from "./BaseStyle";

type Props = {
  /** custom html tag */
  customComponent?: string;
};

const TagElement = ({ customComponent, ...props }: Props) =>
  React.createElement(tag[customComponent || "div"], { ...props });

const { colors, fontSizes } = theme;
// H1
const H1 = styled(TagElement)`
  font-size: ${fontSizes[8]};
  color: ${colors.N90};
  ${BaseStyle};
  ${color};
  ${fontSize};
`;
H1.displayName = "Heading.H1";

// H2
const H2 = styled(TagElement)`
  font-size: ${fontSizes[6]};
  color: ${colors.N90};
  ${BaseStyle};
  ${color};
  ${fontSize};
`;
H2.displayName = "Heading.H2";

// H3
const H3 = styled(TagElement)`
  font-size: ${fontSizes[5]};
  color: ${colors.N90};
  ${BaseStyle};
  ${color};
  ${fontSize};
`;
H3.displayName = "Heading.H3";

// H4
const H4 = styled(TagElement)`
  font-size: ${fontSizes[4]};
  color: ${colors.N90};
  ${BaseStyle};
  ${color};
  ${fontSize};
`;
H4.displayName = "Heading.H4";

// H5
const H5 = styled(TagElement)`
  font-size: ${fontSizes[3]};
  color: ${colors.N70};
  ${BaseStyle};
  ${color};
  ${fontSize};
`;
H5.displayName = "Heading.H5";

// H6
const H6 = styled(TagElement)`
  font-size: ${fontSizes[1]};
  color: ${colors.N70};
  ${BaseStyle};
  ${color};
  ${fontSize};
`;
H6.displayName = "Heading.H6";

class Heading extends React.Component<any> {
  static H1 = H1;
  static H2 = H2;
  static H3 = H3;
  static H4 = H4;
  static H5 = H5;
  static H6 = H6;

  static displayName = "Heading";

  render() {
    const children = React.Children.map(
      this.props.children,
      (child: React.ReactElement<any>) => React.cloneElement(child)
    );

    return <React.Fragment {...this.props}>{children}</React.Fragment>;
  }
}

export default Heading;
