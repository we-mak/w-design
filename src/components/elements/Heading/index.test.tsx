import * as React from "react";
import { render } from "react-testing-library";
import "jest-styled-components";
import Heading from "./index";

describe("<Grid/>", () => {
  // Render test
  it("should renders correctly", () => {
    const { container } = render(
      <>
        <Heading.H1 />
        <Heading.H2 />
        <Heading.H3 />
        <Heading.H4 />
        <Heading.H5 />
        <Heading.H6 />
      </>
    );
    expect(container).toMatchSnapshot();
  });
});
