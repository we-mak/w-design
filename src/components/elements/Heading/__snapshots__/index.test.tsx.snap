// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid/> should renders correctly 1`] = `
.c0 {
  font-size: 2rem;
  display: block;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-weight: 700;
  color: #394B59;
}

.c1 {
  font-size: 1.6rem;
  display: block;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-weight: 700;
  color: #394B59;
}

.c2 {
  font-size: 1.4rem;
  display: block;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-weight: 700;
  color: #394B59;
}

.c3 {
  font-size: 1.2rem;
  display: block;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-weight: 700;
  color: #394B59;
}

.c4 {
  font-size: 1rem;
  display: block;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-weight: 700;
  color: #394B59;
}

.c5 {
  font-size: 0.9rem;
  display: block;
  line-height: 1.25;
  margin-bottom: 0.5em;
  margin-top: 0;
  font-weight: 700;
  color: #394B59;
}

<div>
  <div
    class="c0"
  />
  <div
    class="c1"
  />
  <div
    class="c2"
  />
  <div
    class="c3"
  />
  <div
    class="c4"
  />
  <div
    class="c5"
  />
</div>
`;
